File: app/src/main/java/com/dmi/perfectreader/layout/liner/BreakFinder.java
package com.dmi.perfectreader.layout.liner;

import com.dmi.util.annotation.Reusable;

import java.util.Locale;

import java8.util.function.Consumer;

public interface BreakFinder {
    void findBreaks(CharSequence text, Locale locale, Consumer<Break> consumer);

    @Reusable
    interface Break {
        int index();
        boolean hasHyphen();
        boolean isForce();
    }
}

File: app/src/main/kotlin/com/dmi/perfectreader/layout/liner/BreakFinder.kt
package com.dmi.perfectreader.layout.liner

import com.dmi.util.annotation.Reusable
import java.util.*

interface BreakFinder {
    fun findBreaks(text: CharSequence, locale: Locale, accept: (Break) -> Unit)

    @Reusable
    class Break {
        var index = 0
        var hasHyphen = false
    }
}
